v1/lib/core/utils.js

```diff
@@ -38,53 +38,90 @@ function idx(target, keyPaths) {
   );
 }
 
-function getGitLastUpdated(filepath) {
-  function isTimestamp(str) {
-    return /^\d+$/.test(str);
+const GIT_LAST_UPDATED_TYPE = {
+  TIME: 'time',
+  AUTHOR: 'author',
+};
+
+function getGitLastUpdated(filepath, type) {
+  if (Object.values(GIT_LAST_UPDATED_TYPE).indexOf(type) === -1) {
+    return null;
   }
 
   // Wrap in try/catch in case the shell commands fail (e.g. project doesn't use Git, etc).
   try {
+    const format = GIT_LAST_UPDATED_TYPE.TIME === type ? '%ct' : 'author=%an';
```

Comment:
Do we need the `author` here?

Let's do away with this `format` variable. Instead, we could do

```
git log --follow --summary --format="%ct, %an" <file>
```

which would generate something like:

```
1539553317, Yangshun Tay

1539457591, Yangshun Tay

1539185600, Yangshun Tay

1538339532, Fienny Angelina

1537426859, Marvin Heilemann

1537330146, Yangshun Tay

1537203487, Yangshun Tay

 copy package.json => v1/package.json (81%)
1537169695, Yangshun Tay

1537157761, Yangshun Tay

1537157671, Endilie Y

1537157183, endiliey
```

and then parse each line for the timestamp and author using a regex that extracts the timestamp and the author. Exampe regex here - https://regex101.com/r/O0HQrc/1

This method will just return both the relevant timestamp and author as an object `{ timestamp: 12345, author: 'Fienny' }`, and it's up to the callsite to use whichever field they need.

Reply:
I initially put the author to differentiate author vs commit summary since both are just plain string 😁 Wdyt about making it stronger like `^(\d{10}), (.+)$ ` ?